zerver/actions/custom_profile_fields.py

```diff
@@ -209,6 +216,48 @@ def do_update_user_custom_profile_data_if_changed(
             },
         )
 
+        new_value = get_custom_profile_field_display_value(field_value)
+
+        changes.append(
+            UserProfileChangeDict(
+                field_name=field_value.field.name,
+                old_value=old_value,
+                new_value=new_value,
+            )
+        )
+
+    if changes and notify:
+        send_user_profile_update_notification(
+            user_profile=user_profile, acting_user=acting_user, changes=changes
+        )
+
+
+def get_custom_profile_field_display_value(field_value: CustomProfileFieldValue) -> str:
+    """Convert custom profile field value to human-readable string based on field type."""
+    type = field_value.field.field_type
+
+    if type == CustomProfileField.SELECT:
+        field_data_dict = orjson.loads(field_value.field.field_data)
+        value_key = field_value.value
+        if value_key in field_data_dict:
+            return field_data_dict[value_key]["text"]
+        else:
+            return field_value.value
```

Comment:
I don't think it is possible for `field_value.value` to not be a key of `field_data_dict` since we validate the date before updating the custom profile field value.